### ISSUE TITLE
[Feature request] Specs pricing table

### ISSUE THREAD

**Issue by nsolerieu** [primer/brand#683]
## Problem

The current `comparison table` and `PricingOption` components don't work for our complex spec based feature pricing like Codespaces, Package or Actions. We tried to retrofit these into both existing component, unsatisfyingly

![Image](https://github.com/user-attachments/assets/d7a9989c-d192-4108-8386-9535147f7185)

## Solution

A dedicated `specsPricingTable` component that allow to display tabular data.

![Image](https://github.com/user-attachments/assets/939d0ab0-062a-47ab-98e8-17e4c0e4cc91)

Allow `Row` or `Column` (flow prop?) based display for more flexibility.

![Image](https://github.com/user-attachments/assets/77848d61-b87b-4153-8ea6-7c0fe4221051)

With one structure that can flex across all type of pricing with non-identical axis item and variations in disclaimer and secondary content

![Image](https://github.com/user-attachments/assets/633384b8-4a59-4133-b3eb-1b47c566a5b1)

[See explorations in figma here](https://www.figma.com/design/SSBWMP19m11Fi3tQ7eYHE7/Features?node-id=494-3135&t=kUQwZKd3bCibAeNp-1)

## Urgency

No urgency, the current feature page needing update will be built statically by @simurai (see Site deign EPIC [here](https://github.com/github/brand-marketing-design/issues/1493) )

**Comment by nsolerieu** [comment]
Due to PMM feedback and update no need to upstream this component yet until it’s been tested.- [Full doc here](https://docs.google.com/document/d/1xZJWAJcgyhX5wCSigVivFd6JYMG-DS-Lou2rjGZSkcA/edit)

@simurai static build will provide the opportunity to test assumptions and circle back

Context:
- Packages will be archived and redirected to Actions
- PM’s expressed for each page where a comparison table or pricing component was present that they would like to see pricing located in the docs as it changes often.

TLDR: on-hold for now
